### DISCUSSION TITLE
Service Client Dependency Injection

### DISCUSSION THREAD

**Discussion by mjfpalmer** [microsoft/PowerPlatform-DataverseServiceClient#187] [Ideas]
Is there any plan to expose methods to use ServiceClient using dependency injection and register a ServiceClient as a service in the IServiceCollection in ASP.NET Core applications (similar to Entity Framework's [AddDbContext](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.entityframeworkservicecollectionextensions.adddbcontext?view=efcore-5.0) method)?

**Comment by MattB-msft** [comment]
Yes, there is a plan to do that,  We are _nearly_ at the point where we can do it. 
It will likely show up in 0.6.x or 0.7.x depending on timing.

**Reply by mjfpalmer** [reply]
Great, thanks @MattB-msft!

**Reply by chrispiasecki** [reply]
@MattB-msft any update on this?

**Reply by twiga2013** [reply]
Hi @MattB-msft is taking constructor dependency on ServiceClient supported now? Looking for something similar to EF AddDataContext. Thanks

**Reply by MattB-msft** [reply]
Yes, your safe with that now, 
We do want to publish what we are doing with internally for DI but are not 100% comfortable with publishing it yet.
